### ISSUE TITLE
XIO:  fatal IO error 11

### ISSUE THREAD

**Issue by mgamsjager** [ConSol/docker-headless-vnc-container#115]
**Image**
From git repo. Docker build locally.
**Tag**
095d9bbd7d3a54884f7a272e6f251f91ef72f207
**Short overview**

Starting VNC results in: 
```
Xvnc TigerVNC 1.8.0 - built Nov  2 2018 19:05:14
Copyright (C) 1999-2017 TigerVNC Team and many others (see README.txt)
See http://www.tigervnc.org for information on TigerVNC.
Underlying X server release 12001000, The X.Org Foundation


Thu Jan 24 15:37:47 2019
 vncext:      VNC extension running!
 vncext:      Listening for VNC connections on all interface(s), port 5901
 vncext:      created VNC server for screen 0
Killing Xvnc process ID 105
XIO:  fatal IO error 11 (Resource temporarily unavailable) on X server ":1"
      after 109 requests (109 known processed) with 10 events remaining.
```

**Detailed error description**

After starting the image I could not connect via VNC. Looking at the log files showed me this error:
```
Killing Xvnc process ID 105
XIO:  fatal IO error 11 (Resource temporarily unavailable) on X server ":1"
      after 109 requests (109 known processed) with 10 events remaining.
```

**Additional content**
Used `Dockerfile.centos.xfce.vnc` from the repo.

**Comment by s1hofmann** [comment]
I have to admit that I'm not sure about what might have caused this error.
Did you do another build?

Anyways, as a result, I just added [dgoss](https://github.com/aelsabbahy/goss/tree/master/extras/dgoss) to our [CI build](https://travis-ci.org/ConSol/docker-headless-vnc-container) to continuously check our containers runtime before pushing a new tag.

```
INFO: Running Tests
Port: tcp:5901: listening: matches expectation: [true]
Port: tcp:5901: ip: matches expectation: [["0.0.0.0"]]
File: /headless/wm_startup.sh: exists: matches expectation: [true]
File: /headless/wm_startup.sh: mode: matches expectation: ["0777"]
File: /headless/wm_startup.sh: owner: matches expectation: ["root"]
File: /headless/wm_startup.sh: group: matches expectation: ["root"]
File: /headless/wm_startup.sh: filetype: matches expectation: ["file"]
File: /dockerstartup/vnc_startup.sh: exists: matches expectation: [true]
File: /dockerstartup/vnc_startup.sh: mode: matches expectation: ["0777"]
File: /dockerstartup/vnc_startup.sh: owner: matches expectation: ["root"]
File: /dockerstartup/vnc_startup.sh: group: matches expectation: ["root"]
File: /dockerstartup/vnc_startup.sh: filetype: matches expectation: ["file"]
Process: Xvnc: running: matches expectation: [true]
Port: tcp:6901: listening: matches expectation: [true]
Port: tcp:6901: ip: matches expectation: [["0.0.0.0"]]
HTTP: http://localhost:6901/?password=vncpassword: status: matches expectation: [200]
Total Duration: 0.009s
Count: 16, Failed: 0, Skipped: 0
```

Sorry I can't provide you with more help at the moment.

**Comment by mgamsjager** [comment]
The steps I took (running OSX 10.14.2):

1. Checkout Git repo
2. `docker build -t console-test -f Dockerfile.centos.xfce.vnc .`
3. `docker run -p 5901:5901 -p 6901:6901 console-test`

I can not connect with VNC viewer. 
http://localhost:6901/?password=vncpassword returns a 200 alright but the screens displays : `Something went wrong, connection is close`

Running `docker run -it -p 5901:5901 -p 6901:6901 console-test bash` to see what's going on:
```
bash-4.2$ ps aux
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
default      1  0.4  0.0  17412  3376 pts/0    Ss   20:31   0:00 bash
default     19  0.0  0.0  17276  3064 pts/0    S    20:31   0:00 bash /headless/noVNC/utils/launch.sh --vnc localhost:5901 --listen 6901
default     34  0.8  0.2 170308 24000 pts/0    S    20:31   0:00 /usr/bin/python /headless/noVNC/utils/websockify/run --web /headless/noVNC/utils/../ 6901 localhost:5901
default    108  0.0  0.0  25448   260 ?        Ss   20:31   0:00 /usr/bin/gpg-agent --sh --daemon --write-env-file /headless/.cache/gpg-agent-info
default    117  0.0  0.0  55168  3652 pts/0    R+   20:31   0:00 ps aux
```

Running `vncserver`:
```
bash-4.2$ vncserver

New 'cd45af1b9290:1 ()' desktop is cd45af1b9290:1

Starting applications specified in /headless/.vnc/xstartup
Log file is /headless/.vnc/cd45af1b9290:1.log

bash-4.2$ cat /headless/.vnc/cd45af1b9290\:1.log

Xvnc TigerVNC 1.8.0 - built Nov  2 2018 19:05:14
Copyright (C) 1999-2017 TigerVNC Team and many others (see README.txt)
See http://www.tigervnc.org for information on TigerVNC.
Underlying X server release 12001000, The X.Org Foundation


Thu Jan 24 20:32:37 2019
 vncext:      VNC extension running!
 vncext:      Listening for VNC connections on all interface(s), port 5901
 vncext:      created VNC server for screen 0
Killing Xvnc process ID 125
bash-4.2$
```

Running `/dockerstartup/vnc_startup.sh` results in the same.

**Comment by mgamsjager** [comment]
Looks like the Ubuntu Dockerfile produces an image that does work but both CentOS DockerFiles do not on my machine.

**Comment by geberl** [comment]
Can confirm the issue for the image built from `Dockerfile.centos.xfce.vnc`. I see the same output for `cat /headless/.vnc/bfcf5f4fa27f\:1.log` as posted in the original bug report. `vncserver -list` shows that nothing is running.

And @mgamsjager , I got a solution:
- Login into your container with the default user `docker exec -it my-centos-vnc-container /bin/bash`
- Insert **exec** in line 5 of `/headless/.vnc/xstartup`, so it says `exec /etc/X11/xinit/xinitrc` instead of `/etc/X11/xinit/xinitrc`
- Exit and reboot the container
- `vncserver` now is starting up correctly, `vncserver -list` shows an active session

**Comment by s1hofmann** [comment]
Sorry I messed things up a bit on my side.

You're right, the current master versions of our CentOS images fail to properly start a VNC server.
This is caused by an updated version of TigerVNC in the CentOS repos.
The issue has already been adressed and should be fixed on `dev` branch, see #112.

Could you please do a build from `dev` branch and let me know the outcome?

A new release is scheduled for end of February.

**Comment by geberl** [comment]
👍 building from dev branch works
Also the locale problems I reported in #116 are gone here, maybe this issue can also be closed
